MethodCallExprContext.matchTypeParameters not considered autoboxing

I have a problem show below.
Not considered java autoboxing?

```
Exception in thread "main" java.lang.RuntimeException: Unable to calculate the type of a parameter of a method call. Method call: System.out.println(Demob.demob(userId)), Parameter: Demob.demob(userId)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveArguments(JavaParserFacade.java:243)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:258)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:131)
	at com.github.javaparser.symbolsolver.JavaSymbolSolver.resolveDeclaration(JavaSymbolSolver.java:161)
	at com.github.javaparser.ast.expr.MethodCallExpr.resolve(MethodCallExpr.java:313)
	at zhibo.tools.upgrade.JavaApp$1.visit(JavaApp.java:31)
	at zhibo.tools.upgrade.JavaApp$1.visit(JavaApp.java:1)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.expr.MethodCallExpr.accept(MethodCallExpr.java:115)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:431)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.stmt.ExpressionStmt.accept(ExpressionStmt.java:71)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:1070)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.NodeList.accept(NodeList.java:255)
	at com.github.javaparser.ast.visitor.ModifierVisitor.modifyList(ModifierVisitor.java:1109)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:169)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.stmt.BlockStmt.accept(BlockStmt.java:70)
	at com.github.javaparser.ast.visitor.ModifierVisitor.lambda$visit$56(ModifierVisitor.java:620)
	at com.github.javaparser.ast.visitor.ModifierVisitor$$Lambda$180/1613255205.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:620)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.body.MethodDeclaration.accept(MethodDeclaration.java:100)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:1070)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.NodeList.accept(NodeList.java:255)
	at com.github.javaparser.ast.visitor.ModifierVisitor.modifyList(ModifierVisitor.java:1109)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:241)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.body.ClassOrInterfaceDeclaration.accept(ClassOrInterfaceDeclaration.java:94)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:1070)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.NodeList.accept(NodeList.java:255)
	at com.github.javaparser.ast.visitor.ModifierVisitor.modifyList(ModifierVisitor.java:1109)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:281)
	at com.github.javaparser.ast.visitor.ModifierVisitor.visit(ModifierVisitor.java:51)
	at com.github.javaparser.ast.CompilationUnit.accept(CompilationUnit.java:128)
	at zhibo.tools.upgrade.JavaApp.main(JavaApp.java:28)
Caused by: java.lang.UnsupportedOperationException: com.github.javaparser.resolution.types.ResolvedPrimitiveType
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.matchTypeParameters(MethodCallExprContext.java:373)
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.resolveMethodTypeParameters(MethodCallExprContext.java:349)
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.solveMethodAsUsage(MethodCallExprContext.java:93)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveMethodAsUsage(JavaParserFacade.java:611)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:289)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:66)
	at com.github.javaparser.ast.expr.MethodCallExpr.accept(MethodCallExpr.java:115)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:475)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:337)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveArguments(JavaParserFacade.java:239)
	... 39 more
```

```java
public class Demo {

	public void demo1(long userId,long roomId) {
		demo2(userId);
		demo2(userId,roomId);
		System.out.println(demo2(userId));
		System.out.println(demo2(Long.valueOf(userId)));
		//
		Demob.demob(userId);
		Demob.demob(userId,roomId);
		System.out.println(Demob.demob(Long.valueOf(userId)));
                //there was an error when resovle this MethodCallExpr 
		System.out.println(Demob.demob(userId));
	}
	
	public static <T>T demo2(T...objs){
		return null;
	}
}
```
```java
public class Demob {

	public static <T>T demob(T...objs) {
		return null;
	}
	public <T>List<T> j2(T...objs) {
		return null;
	}
}
```
```java
public class JavaApp {

	public static void main(String[] args) throws FileNotFoundException {
		final CombinedTypeSolver typeSolver = new CombinedTypeSolver();
		typeSolver.add(new ReflectionTypeSolver());
		typeSolver.add(new MemoryTypeSolver());
		typeSolver.add(new JavaParserTypeSolver("xxx\\src\\main\\java\\"));
		JavaSymbolSolver symbolSolver = new JavaSymbolSolver(typeSolver);
		StaticJavaParser.getConfiguration().setSymbolResolver(symbolSolver);
		CompilationUnit cu = StaticJavaParser.parse(new File("xxx\\Demo.java"));
		cu.accept(new ModifierVisitor<Void>() {
			@Override
			public Visitable visit(MethodCallExpr n, Void arg) {
				ResolvedMethodDeclaration resolve = n.resolve();
				System.out.println(resolve.getQualifiedSignature());
				return super.visit(n, arg);
			}
		}, null);
	}
}
```

--- a/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/declarations/common/MethodDeclarationCommonLogic.java
+++ b/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/declarations/common/MethodDeclarationCommonLogic.java
@@ -23,6 +23,7 @@ package com.github.javaparser.symbolsolver.declarations.common;
 
 import com.github.javaparser.resolution.MethodUsage;
 import com.github.javaparser.resolution.declarations.ResolvedMethodDeclaration;
+import com.github.javaparser.resolution.declarations.ResolvedParameterDeclaration;
 import com.github.javaparser.resolution.declarations.ResolvedTypeParameterDeclaration;
 import com.github.javaparser.resolution.types.ResolvedType;
 import com.github.javaparser.resolution.types.ResolvedTypeVariable;
@@ -60,9 +61,15 @@ public class MethodDeclarationCommonLogic {
         // and then we replace them in the return type
         // Map<TypeParameterDeclaration, Type> determinedTypeParameters = new HashMap<>();
         InferenceContext inferenceContext = new InferenceContext(MyObjectProvider.INSTANCE);
-        for (int i = 0; i < methodDeclaration.getNumberOfParams() - (methodDeclaration.hasVariadicParameter() ? 1 : 0); i++) {
-            ResolvedType formalParamType = methodDeclaration.getParam(i).getType();
+        for (int i = 0; i < methodDeclaration.getNumberOfParams(); i++) {
+            ResolvedParameterDeclaration formalParamDecl = methodDeclaration.getParam(i);
+            ResolvedType formalParamType = formalParamDecl.getType();
             ResolvedType actualParamType = parameterTypes.get(i);
+
+            if (formalParamDecl.isVariadic() && !actualParamType.isArray()) {
+                formalParamType = formalParamType.asArrayType().getComponentType();
+            }
+
             inferenceContext.addPair(formalParamType, actualParamType);
         }
 

--- a/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/GenericsResolutionTest.java
+++ b/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/GenericsResolutionTest.java
@@ -242,6 +242,20 @@ class GenericsResolutionTest extends AbstractResolutionTest {
     }
 
     @Test
+    void resolveUsageOfMethodOfGenericClassWithBoxing() {
+        CompilationUnit cu = parseSample("Generics");
+        ClassOrInterfaceDeclaration clazz = Navigator.demandClass(cu, "GenericMethodBoxing");
+        MethodDeclaration method = Navigator.demandMethod(clazz, "bar");
+        MethodCallExpr expression = Navigator.findMethodCall(method, "foo").get();
+
+        MethodUsage methodUsage = JavaParserFacade.get(new ReflectionTypeSolver()).solveMethodAsUsage(expression);
+
+        assertEquals("foo", methodUsage.getName());
+        assertEquals("GenericMethodBoxing", methodUsage.declaringType().getName());
+        assertEquals("java.lang.Long", methodUsage.returnType().describe());
+    }
+
+    @Test
     void resolveElementOfList() {
         CompilationUnit cu = parseSample("ElementOfList");
         ClassOrInterfaceDeclaration clazz = Navigator.demandClass(cu, "ElementOfList");

--- a/javaparser-symbol-solver-testing/src/test/resources/Generics.java.txt
+++ b/javaparser-symbol-solver-testing/src/test/resources/Generics.java.txt
@@ -49,3 +49,15 @@ public final class GenericMethodCalls {
     }
 
 }
+
+public final class GenericMethodBoxing {
+
+    public <T> T foo(T... t) {
+        return null;
+    }
+
+    public void bar() {
+        foo(5L);
+    }
+
+}

